spot-client/src/common/app-state/setup/reducer.test.js

```diff
@@ -0,0 +1,75 @@
+import * as types from './action-types';
+import setupReducer from './reducer';
+
+describe('setup reducer', () => {
+    it('by default has setup as not complete', () => {
+        expect(setupReducer(undefined, {}))
+            .toEqual(expect.objectContaining({ completed: false }));
+    });
+
+    it('sets an avatar url', () => {
+        const avatarUrl = 'new-url';
+
+        expect(setupReducer(undefined, {
+            type: types.SET_AVATAR_URL,
+            avatarUrl
+        }))
+        .toEqual(expect.objectContaining({ avatarUrl }));
+    });
+
+    it('sets a display name', () => {
+        const displayName = 'new-name';
+
+        expect(setupReducer(undefined, {
+            type: types.SET_DISPLAY_NAME,
+            displayName
+        }))
+        .toEqual(expect.objectContaining({ displayName }));
+    });
+
+    it('sets a jwt', () => {
+        const jwt = 'test-jwt';
+
+        expect(setupReducer(undefined, {
+            type: types.SET_JWT,
```

Comment:
I'm too lazy to look at the code, but don't we have action functions ? Would it not be better to use that actions instead ? This duplicates the code a little bit.

Reply:
It's weird to me for a unit test but also makes sense to test together to hide implementation details. action creators could be used to fire reducers and then selectors used. This might require initializing a redux store with each test, but maybe that's okay.

Reply:
I'm converting these to integration tests.